mopidy_tidal/library.py

```diff
@@ -1,155 +1,37 @@
-        return getattr(self._session, item_type, None)
-
-    def _get_images(self, uri) -> List[Image]:
-        assert uri.startswith("tidal:"), f"Invalid TIDAL URI: {uri}"
-
-        parts = uri.split(":")
-        item_type = parts[1]
-        if item_type == "track":
-            # For tracks, retrieve the artwork of the associated album
-            item_type = "album"
-            item_id = parts[3]
-            uri = ":".join([parts[0], "album", parts[3]])
-        elif item_type == "album":
-            item_id = parts[2]
-        elif item_type == "playlist":
-            item_id = parts[2]
-        elif item_type == "artist":
-            item_id = parts[2]
-        elif item_type == "mix":
-            item_id = parts[2]
-        else:
-            # uri has no image associated to it (eg. tidal:mood tidal:genres etc.)
-            return []
-
-        if uri in self._image_cache:
-            # Cache hit
-            return self._image_cache[uri]
-
-        logger.debug("Retrieving %r from the API", uri)
-        getter = self._get_api_getter(item_type)
-        if not getter:
-            logger.warning("The API item type %s has no session getters", item_type)
-            return []
-
-        item = getter(item_id)
-        if not item:
-            logger.debug("%r is not available on the backend", uri)
-            return []
-
-        img_uri = self._get_image_uri(item)
-        if not img_uri:
-            logger.debug("%r has no associated images", uri)
-            return []
-
-        logger.debug("Image URL for %r: %r", uri, img_uri)
-        return [Image(uri=img_uri, width=320, height=320)]
-
-    def __call__(self, uri: str) -> Tuple[str, List[Image]]:
-        try:
-            return uri, self._get_images(uri)
-        except (AssertionError, AttributeError, HTTPError) as err:
-            logger.error("%s when processing URI %r: %s", type(err), uri, err)
-            return uri, []
-
-    def cache_update(self, images):
-        self._image_cache.update(images)
-
-
 class TidalLibraryProvider(backend.LibraryProvider):
-    root_directory = models.Ref.directory(uri="tidal:directory", name="Tidal")
-    backend: "TidalBackend"
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._artist_cache = LruCache()
-        self._album_cache = LruCache()
-        self._track_cache = LruCache()
-        self._playlist_cache = PlaylistMetadataCache()
-
-    @property
-    def _session(self):
-        return self.backend.session
+    root_directory = Ref.directory(uri=str(URI(URIType.DIRECTORY)), name="Tidal")
 
-    @login_hack(passthrough=True)
-    def get_distinct(self, field, query=None) -> set[str]:
+    def get_distinct(self, field, query=None):
         from mopidy_tidal.search import tidal_search
 
-        logger.debug("Browsing distinct %s with query %r", field, query)
-        session = self._session
+        logger.info(f"Browsing distinct {field!s} with query {query!r}")
```

Comment:
Nitpicking, but I think (at least according to Black's style conventions) the `log.info("msg: %s", var)` notation is preferred over `log.info(f"msg: {msg}")` - at least for logging functions.

Reply:
(the reason for this btw is that f-strings are eagerly evaluated, but a logging function might never produce any output. Using `%` pushes the formatting down the stack and (in theory) avoids doing it if the log level suppresses the output)

Reply:
That is correct. Don't consider this change. I was trying to extract only the model mapping change and other things were included. And I was wrong to change them, anyway.